File: erlang/rna-transcription/rna_transcription.erl
-module(rna_transcription).
-export([to_rna/1]).

to_rna([H|T]) -> [to_rna(H)|to_rna(T)];
to_rna([]) -> [];
to_rna($G) -> $C;
to_rna($C) -> $G;
to_rna($T) -> $A;
to_rna($A) -> $U.

File: elixir/rna-transcription/lib/rna_transcription.ex
defmodule RnaTranscription do
  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

      iex> RnaTranscription.to_rna('ACTG')
      'UGAC'

  """
  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    dna
    |> Enum.map(
      &case &1 do
        ?G ->
          ?C

        ?T ->
          ?A

        ?A ->
          ?U

        ?C ->
          ?G
      end
    )
  end
end
